From 14069b162a3e17775af2a2bcf0dc573b8edc7901 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@freescale.com>
Date: Wed, 9 Dec 2015 18:51:22 +0800
Subject: [PATCH] MLK-11985-1 ARM: imx: fix coverity scan issues

1. Add return value check to fix below warning:

check_return: Calling of_property_read_u32 without
checking return value.

2. Improve code to fix below warning:

copy_paste_error: i in of_clk_get(dev->of_node, i)
looks like a copy-paste error.

Signed-off-by: Anson Huang <Anson.Huang@freescale.com>
---
 arch/arm/mach-imx/gpc.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-imx/gpc.c b/arch/arm/mach-imx/gpc.c
index 6ee4127..39651dc 100644
--- a/arch/arm/mach-imx/gpc.c
+++ b/arch/arm/mach-imx/gpc.c
@@ -657,12 +657,12 @@ static int imx_gpc_genpd_init(struct device *dev, struct regulator *pu_reg)
 	ipg = of_clk_get(dev->of_node, pu_clks);
 
 	/* Get disp domain clks */
-	for (k = 0, i = pu_clks + ipg_clks; i < pu_clks + ipg_clks + disp_clks;
-		i++, k++) {
+	for (i = pu_clks + ipg_clks; i < pu_clks + ipg_clks + disp_clks;
+		i++) {
 		clk = of_clk_get(dev->of_node, i);
 		if (IS_ERR(clk))
 			break;
-		imx6s_display_domain.clk[k] = clk;
+		imx6s_display_domain.clk[k++] = clk;
 	}
 	imx6s_display_domain.num_clks = k;
 
@@ -691,7 +691,10 @@ static int imx_gpc_probe(struct platform_device *pdev)
 	struct regulator *pu_reg;
 	int ret;
 
-	of_property_read_u32(pdev->dev.of_node, "fsl,ldo-bypass", &bypass);
+	if (of_property_read_u32(pdev->dev.of_node, "fsl,ldo-bypass", &bypass))
+		dev_warn(&pdev->dev,
+			"no fsl,ldo-bypass found!\n");
+
 	pu_reg = devm_regulator_get(&pdev->dev, "pu");
 	if (!IS_ERR(pu_reg)) {
 		/* The regulator is initially enabled */
-- 
1.8.0

